receiver/haproxyreceiver/scraper.go

```diff
@@ -15,22 +15,86 @@
 package haproxyreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/haproxyreceiver"
 
 import (
+	"bytes"
 	"context"
+	"encoding/csv"
+	"net"
+	"strings"
+	"time"
 
-	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/haproxyreceiver/internal/metadata"
 )
 
 type scraper struct {
+	endpoint       string
+	logger         *zap.Logger
+	metricsBuilder *metadata.MetricsBuilder
 }
 
-func (s scraper) Scrape(ctx context.Context) (pmetric.Metrics, error) {
-	return pmetric.NewMetrics(), nil
+func (s *scraper) scrape(_ context.Context) (pmetric.Metrics, error) {
+
+	c, err := net.Dial("unix", s.endpoint)
+	if err != nil {
+		return pmetric.NewMetrics(), err
+	}
+	defer c.Close()
+	records, err := s.readStats(c)
+	if err != nil {
+		return pmetric.NewMetrics(), err
+	}
+
+	now := pcommon.NewTimestampFromTime(time.Now())
+	for _, record := range records {
+		err = s.metricsBuilder.RecordHaproxySessionsCountDataPoint(now, record["scur"], record["pxname"], record["svname"])
+		if err != nil {
+			return pmetric.NewMetrics(), err
+		}
+	}
+
+	return s.metricsBuilder.Emit(metadata.WithHaproxyAddr(s.endpoint)), nil
+}
+
+func (s *scraper) readStats(c net.Conn) ([]map[string]string, error) {
+	_, err := c.Write([]byte("show stats\n"))
```

Comment:
If this is a fixed byte array, it might be worth making this a global var (since array's can't be static, or a cast to a modifiable type)

Reply:
ok I'll do that.